jsonrpc/server_test.go

```diff
@@ -453,6 +493,22 @@ func TestHandle(t *testing.T) {
 			req:     `[1,2,3]`,
 			res:     `[{"jsonrpc":"2.0","error":{"code":-32600,"message":"Invalid Request","data":"json: cannot unmarshal number into Go value of type jsonrpc.request"},"id":null},{"jsonrpc":"2.0","error":{"code":-32600,"message":"Invalid Request","data":"json: cannot unmarshal number into Go value of type jsonrpc.request"},"id":null},{"jsonrpc":"2.0","error":{"code":-32600,"message":"Invalid Request","data":"json: cannot unmarshal number into Go value of type jsonrpc.request"},"id":null}]`,
 		},
+		"echo value": {
+			req: `{"jsonrpc": "2.0", "method": "echoValue", "params": [10], "id": 1}`,
+			res: `{"jsonrpc":"2.0","result":10,"id":1}`,
+		},
+		"echo error": {
+			req: `{"jsonrpc": "2.0", "method": "echoError", "params": ["echo"], "id": 1}`,
+			res: `{"jsonrpc":"2.0","error":{"code":44,"message":"echo"},"id":1}`,
+		},
+		"echo nil error": {
+			req: `{"jsonrpc": "2.0", "method": "echoError", "params": ["nil"], "id": 1}`,
+			res: `{"jsonrpc":"2.0","id":1}`,
```

Comment:
I believe `result` is a required member of the response. ([spec](https://www.jsonrpc.org/specification#response_object))

Reply:
You are right, [result should be null](https://github.com/ethereum/go-ethereum/blob/master/rpc/testdata/reqresp-noargsrets.js)